sdk/formrecognizer/ai-form-recognizer/test/public/training.spec.ts

```diff
@@ -35,11 +37,21 @@ matrix(
       let recorder: Recorder;
 
       beforeEach(async function (this: Context) {
+        if (!isNode && useAad && getAudience() !== FormRecognizerAudience.AzurePublicCloud) {
+          // Skipping the browser AAD tests with non-public clouds since the Client Secret Credential fails with
```

Comment:
Let's mark this with a TODO to investigate a little more deeply and decide if this is possible or not. Only reason being it's easy to search for instances of TODO later when we're preparing to release this as stable. At that point we can decide whether or not we want to try to test this corner case.

Reply:
The error `Confidential Client is not supported in Cross Cloud request` is one I have commonly seen when the wrong tenant id and subscription id are entered (e.g. either the cloud specific tenant or subscription isn't properly passed everywhere).

Reply:
Oh, upon discussion with @sadasant, we assumed it to be an AAD issue.

Reply:
The credentials used are being properly passed through karma.conf, 
![image](https://user-images.githubusercontent.com/10452642/156639226-2bf834ad-9e1e-40b0-aed1-2ad2c3e66a7d.png)

So, we do have the proper secrets in the browser I believe.

Reply:
The error: `Confidential Client is not supported in Cross Cloud request` appears because these tests are using the `ClientSecretCredential` in the browser.

This credential is really a bad pattern, since it’s not really how browser code will work. `ClientSecretCredential` for the browser is kept for backwards compatibility, but the only way to use this credential in the browser is to disable browser security. The only true browser credential is the `InteractiveBrowserCredential` at the moment, and we can’t have automated tests with this credential yet (perhaps in the future we will with Playwright).

If this error does not happen in node but it does happen in the browser, I would believe that the service does not like browser requests 🤔

does this service support CORS requests even? Like, is it possible to use this client with the InteractiveBrowserCredential?

Reply:
Correction: This only occurs for browser tests in non-public clouds.

Reply:
But does it also happen with `InteractiveBrowserCredential`? (you’d have to try manually)

Reply:
If it doesn’t happen with `InteractiveBrowserCredential` I wouldn’t bother bringing this up to the service team, since it’s a really weird ask. But if browser is supposed to work, and we can’t get it to work on non-public clouds, then that’s a concrete ask to the service team.

Reply:
Haven't tried, I guess we can park this here and investigate further in the issue logged.

Reply:
@HarshaNalluru if you're not also passing through `AZURE_SUBSCRIPTION_ID` then it might be using the sovereign cloud AAD tenant, but a default public cloud subscription.

Reply:
Okay, will add the subscription-id to the yml file as agreed here https://github.com/Azure/azure-sdk-for-js/pull/20522#discussion_r819097862, though it should have no impact. (Just for consistency)

Reply:
@sadasant are you certain that the error is because of ClientSecretCredential? We test with CSC in the Azure public cloud frequently. It doesn't make any sense to me that it would work in public but not in the sovereign clouds.

My reading of the error message makes it seem like _cross-cloud_ is the primary factor in the error and not "confidential client", and what @benbp is suggesting makes more sense to me that it's about a mismatch between where the ID tenant/subscription and the FR resource reside.

Reply:
@witemple-msft while this is likely an error in ClientSecretCredential, the copy of ClientSecretCredential used is deprecated: it’s the legacy browser version of it, which is not intended to be used from the browsers. We didn’t remove it back then becase we didn’t have a better approach to test our clients in the browser.